### DISCUSSION TITLE
How to create WaferMap using XYPlot(X,Y,Z)

### DISCUSSION THREAD

**Discussion by munnad31** [jfree/jfreechart#297] [Q&A]
Hello  - I'm looking for sample code to generate a contour plot using a CSV file and Z values should plot dots and color the wafer map (R,G,B) ..Please help

**Comment by trashgod** [comment]
[`XYBlockRenderer`](https://www.jfree.org/jfreechart/javadoc/org/jfree/chart/renderer/xy/XYBlockRenderer.html), mentioned [here](https://stackoverflow.com/search?tab=votes&q=%5bjava%5d%20XYBlockRenderer) and [here](https://www.jfree.org/jfreechart/samples.html), might be worth a look.

For discussion sake, can clarify what is "deprecated using `XYPlot`?"

**Reply by munnad31** [reply]
@trashgod  : I mean there is no more contourplot right its deprecated and moved to XYPlot , I did not frame it correctly ..Thanks

**Reply by munnad31** [reply]
@trashgod : I have upgraded the JFreechart version to 1.5.3 today and in External Libraries in eclipse I placed the Latest Jar and my IDE is not recognizing the imports and I checked the JDK is pointed to 1.8 ...Any Idea on how to fix it ?

**Reply by trashgod** [reply]
Ah, `ContourPlot`, removed [here](https://github.com/jfree/jfreechart/commit/07059fb93ed38147f8ded82ec8d1bf5400be1952).  See also [_Migration from JFreeChart 1.0.x_](https://github.com/jfree/jfreechart#migration-from-jfreechart-10x).
